qark/plugins/broadcast/send_broadcast_receiver_permission.py

```diff
@@ -0,0 +1,196 @@
+    "can potentially receive this broadcast. You should investigate this for potential data leakage."
+)
+STICKY_BROADCAST = (
+    "A sticky broadcast, {broadcast_type}, is sent from this class: {file_name}. These should not be used, as they "
+    "rovide no security (anyone can access them), no protection (anyone can modify them), and many other problems. "
+    "For more info: http://developer.android.com/reference/android/content/Context.html"
+)
+
+
+class SendBroadcastReceiverPermission(BasePlugin):
+    """
+    This plugin checks certain broadcast methods to see if they are using an insecure version,
+    based on number of arguments.
+    """
+    def __init__(self):
+        BasePlugin.__init__(self, category="broadcast")
+        self.severity = Severity.WARNING
+        self.current_file = None
+        self.manifest_xml = None
+        self.below_min_sdk_21 = False
+
+    def run(self, files, apk_constants=None):
+        try:
+            self.below_min_sdk_21 = apk_constants["min_sdk"] < 21
+        except (KeyError, TypeError):
+            for decompiled_file in files:
+                if decompiled_file.lower().endswith("androidmanifest.xml"):
+                    self.below_min_sdk_21 = get_min_sdk(manifest_xml=minidom.parse(decompiled_file)) < 21
+                else:
+                    self.below_min_sdk_21 = False
+
+        java_files = (decompiled_file for decompiled_file in files if decompiled_file.lower().endswith(".java"))
+        for java_file in java_files:
+            with open(java_file, "r") as java_file_to_read:
+                file_contents = java_file_to_read.read()
+
+                # really simple check to see if the file has the methods we are interested in before parsing the AST
+                if not re.search("({broadcasts})".format(broadcasts="|".join(BROADCAST_METHODS)), file_contents):
+                    continue
+
+            try:
+                parsed_tree = javalang.parse.parse(file_contents)
+            except javalang.parser.JavaSyntaxError:
+                log.debug("Error parsing file %s, continuing", java_file)
+                continue
+            self.current_file = java_file
+            for method_invocation in parsed_tree.filter(MethodInvocation):
+                self._check_method_invocation(method_invocation, parsed_tree.imports)
+
+    def _check_method_invocation(self, method_invocation, imports):
+        """
+        Checks the type of method that is being called and the amount of arguments and adds it to the issues list if it
+        results in a warning or vulnerability.
+
+        :param javalang.tree.MethodInvocation method_invocation: method being called
+        :param list imports: list of imports
+        """
+        method_name = method_invocation[1].member
+        num_arguments = len(method_invocation[1].arguments)
+        if method_invocation[1].member in BROADCAST_METHODS:
```

Comment:
1) use method_name, not method_invocation[1].member
2) check for the inverse (not in BROADCAST_METHODS) and return immediately - saves a level of nesting